client/src/app/users/add-user.component.spec.ts

```diff
@@ -132,14 +126,27 @@ describe('AddUserComponent', () => {
       ageControl = addUserComponent.addUserForm.controls.age;
     });
 
-    it('should not allow empty names', () => {
+    it('should not allow empty ages', () => {
       ageControl.setValue('');
       expect(ageControl.valid).toBeFalsy();
     });
 
-    it('should be fine with "27"', () => {
-      ageControl.setValue('27');
-      expect(ageControl.valid).toBeTruthy();
+    describe('should be fine with "27"', () => {
+      // Both the form control and the HTML itself enforce certain constraints
+      // what we can put in the age field. The HTML makes sure that the value
+      // is numeric, and the form control makes sure that it's in the right
+      // range.
+      it('...the form control', () => {
+        ageControl.setValue('27');
+        expect(ageControl.valid).toBeTruthy();
+      });
+
+      it('...and the input field itself', () => {
```

Comment:
Should/can we separate these two tests? Would it make sense to test the form field as a separate set of tests?

That probably shouldn't block this PR - maybe it's a new issue, or maybe it just doesn't make sense. 

Thoughts?

Reply:
Oh, sure—I could move the HTMLy tests into a separate describe block all by themselves, away from the form control tests. Would you prefer that?

Reply:
(I think I'm going to go ahead and separate them out!)